addons/website_sale_product_configurator/tests/test_website_sale_configurator.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import tagged
+from odoo.addons.sale_product_configurator.tests.common import TestProductConfiguratorCommon
+from odoo.addons.base.tests.common import HttpCaseWithUserPortal
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteSaleProductConfigurator(TestProductConfiguratorCommon, HttpCaseWithUserPortal):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_product_custo_desk.write({
+            'optional_product_ids': [(4, cls.product_product_conf_chair.id)],
+            'website_published': True,
+        })
+
+        ptav_ids = cls.product_product_custo_desk.attribute_line_ids.product_template_value_ids
+        ptav_ids.filtered(lambda ptav: ptav.name == 'Aluminium').price_extra = 50.4
+
+    def test_01_product_configurator_variant_price(self):
+        ptav_ids = self.product_product_custo_desk.attribute_line_ids.product_template_value_ids
+        parent_combination = ptav_ids.filtered(lambda ptav: ptav.name in ('Aluminium', 'White'))
+        self.assertEqual(self.product_product_conf_chair._is_add_to_cart_possible(parent_combination), True)
```

Comment:
This should be tested with the portal user since we run the tour as portal user.
```suggestion
        self.assertEqual(self.product_product_conf_chair.with_user(self.user_portal)._is_add_to_cart_possible(parent_combination), True)
```
Doing this reveals that we actually don't have access to the chair product because it is not published.